### DISCUSSION TITLE
Static type checking not working when creating new entities

### DISCUSSION THREAD

**Discussion by shahidcodes** [mikro-orm/mikro-orm#2544] [Q&A]
I've this entity

```ts

@Entity()
export class Truck extends BaseEntity {
  @Property()
  plateNumber!: string;

  @Property()
  sequenceNumber!: string;

  @Property()
  status!: StatusEnum;
}

```

then in my service when I'm trying to create an entry like this 
![image](https://user-images.githubusercontent.com/17620737/146477613-627f5dd0-f6f6-4d44-af5a-9ac95ee1055b.png)

properties are showing as optional not required as defined in the entity class.

I am very new to mikro orm, recently switch from mongoose.

**Comment by B4nan** [comment]
This is already implemented in v5. Also note that they will all be optional, that's how it works, as that method can accept incomplete entity data (e.g. you don't need `id` but that is usually defined as required field).

**Reply by shahidcodes** [reply]
Is there any specific way to have typescript yell at me if I am missing any required field when I create an entity? Or should I validate the input before creating the entity?
I'm using `MongoDB` as driver.

**Reply by B4nan** [reply]
Not really, for runtime validation you can subscribe here #2176, I am not interested in doing compile time validation for the reason I already mentioned above - the PK would be required this way, which does not make sense. If you want this, you need to create your own wrapper over `em.create()` and make your own stricter implementation of the type.

**Reply by shahidcodes** [reply]
I do have runtime validation, but I would love to have compile type checking right from the entity definition along with runtime validations. It makes sense for SQL since it can be done on a database level.
